signer/src/stacks/contracts.rs

```diff
@@ -277,6 +280,10 @@ impl AsContractCall for CompleteDepositV1 {
     ///    deposit request.
     /// 6. That the amount to mint does not exceed the deposit amount.
     /// 7. That the fee is less than the desired max-fee.
+    /// 8. That the fee matches the expected assessed fee for the output.
+    /// 9. That the first input into the sweep transaction is the signers'
+    ///    UTXO. This checks that the sweep transaction was generated by
+    ///    the signers.
```

Comment:
Should we also add this check when fetching the signers UTXO? Otherwise, looking at the constraints in https://github.com/stacks-network/sbtc/issues/538, if someone directly pays `scriptPubKey` (is it possible?) then that would become the new UTXO. (not related to this PR, if needed I'll open an issue)

Reply:
Yes (!), we need that check there too. Anyone can spend to a `scriptPubKey` so we need to check one of the inputs. I had forgotten about that constraint when writing that ticket. I'll open a new one to add that in.

Reply:
Created https://github.com/stacks-network/sbtc/issues/603 for this.